File: tests/runtime/many_arguments/host.py
from exports.bindings import Exports
from imports.bindings import add_imports_to_linker, Imports
import math;
import sys
import wasmtime

class MyImports:
    def many_arguments(self,
            a1: int,
            a2: int,
            a3: int,
            a4: int,
            a5: int,
            a6: int,
            a7: int,
            a8: int,
            a9: int,
            a10: int,
            a11: int,
            a12: int,
            a13: int,
            a14: int,
            a15: int,
            a16: int,
            a17: int,
            a18: int,
            a19: int,
            a20: int) -> None:
        assert(a1 == 1)
        assert(a2 == 2)
        assert(a3 == 3)
        assert(a4 == 4)
        assert(a5 == 5)
        assert(a6 == 6)
        assert(a7 == 7)
        assert(a8 == 8)
        assert(a9 == 9)
        assert(a10 == 10)
        assert(a11 == 11)
        assert(a12 == 12)
        assert(a13 == 13)
        assert(a14 == 14)
        assert(a15 == 15)
        assert(a16 == 16)
        assert(a17 == 17)
        assert(a18 == 18)
        assert(a19 == 19)
        assert(a20 == 20)


def run(wasm_file: str) -> None:
    store = wasmtime.Store()
    module = wasmtime.Module.from_file(store.engine, wasm_file)
    linker = wasmtime.Linker(store.engine)
    linker.define_wasi()
    wasi = wasmtime.WasiConfig()
    wasi.inherit_stdout()
    wasi.inherit_stderr()
    store.set_wasi(wasi)

    imports = MyImports()
    add_imports_to_linker(linker, store, imports)
    wasm = Exports(store, linker, module)

    wasm.many_arguments(store, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,14, 15, 16, 17, 18, 19, 20)

if __name__ == '__main__':
    run(sys.argv[1])

File: tests/runtime/many_arguments/host.ts
import { instantiate } from "./many_arguments.js";
import { loadWasm, testwasi } from "./helpers.js";

function assertEq(x: any, y: any) {
  if (x !== y)
    throw new Error(`${x} != ${y}`);
}

function assert(x: boolean) {
  if (!x)
    throw new Error("assert failed");
}

async function run() {
  const wasm = await instantiate(loadWasm, {
    testwasi,
    imports: {
      manyArguments(
        a1,
        a2,
        a3,
        a4,
        a5,
        a6,
        a7,
        a8,
        a9,
        a10,
        a11,
        a12,
        a13,
        a14,
        a15,
        a16,
      ) {
        assertEq(a1, 1n);
        assertEq(a2, 2n);
        assertEq(a3, 3n);
        assertEq(a4, 4n);
        assertEq(a5, 5n);
        assertEq(a6, 6n);
        assertEq(a7, 7n);
        assertEq(a8, 8n);
        assertEq(a9, 9n);
        assertEq(a10, 10n);
        assertEq(a11, 11n);
        assertEq(a12, 12n);
        assertEq(a13, 13n);
        assertEq(a14, 14n);
        assertEq(a15, 15n);
        assertEq(a16, 16n);
      },
    },
  });

  wasm.manyArguments(
    1n,
    2n,
    3n,
    4n,
    5n,
    6n,
    7n,
    8n,
    9n,
    10n,
    11n,
    12n,
    13n,
    14n,
    15n,
    16n,
  );
}

await run()
